tests/basic_tests.cpp

```diff
@@ -396,3 +396,9 @@ TYPED_TEST(basic_tests, nodejs_50235) {
   ASSERT_EQ(out->get_href(), "http://test.com:5/path?param=1");
   SUCCEED();
 }
+
+// https://github.com/nodejs/node/issues/51514
+TYPED_TEST(basic_tests, nodejs_51514) {
+  auto out = ada::parse<ada::url>("http://1.1.1.256");
```

Comment:
We need to also fix and test this on url_aggregator

```suggestion
  auto out = ada::parse<TypeParam>("http://1.1.1.256");
```

Reply:
Ok, I will fix it too.

Reply:
I have fixed and tested it on `url_aggregator`.

![image](https://github.com/ada-url/ada/assets/42082890/45674f8a-0f41-4191-a77f-db01f50a67ab)